### DISCUSSION TITLE
Network configuration for Helm charts?

### DISCUSSION THREAD

**Discussion by gnometsunami** [concourse/concourse#7419] [Help & Support]
I have installed concourse on kubernetes using the helm charts, but networking seems broken. When running the "hello world" pipeline from the documentation, I am receiving this error:
`ERRO[0006] checking origin busybox failed: initialize transport: Get "https://index.docker.io/v2/": dial tcp 34.193.45.8:443: connect: no route to host `

What steps are missing to get concourse working?

**Comment by taylorsilva** [comment]
The workers probably need DNS configured. See this section of the docs: https://concourse-ci.org/concourse-worker.html#troubleshooting-and-fixing-dns-resolution

You're probably using the Guardian runtime, its the default, so I'll link to those sections of `values.yaml`.

You can try enabling the DNS proxy first. That'll be the easiest thing to do: https://github.com/concourse/concourse-chart/blob/016fb3ea9fcda36f141f2a17316736ca78d01e07/values.yaml#L1663

If that doesn't work then you can try setting specific DNS servers here: https://github.com/concourse/concourse-chart/blob/016fb3ea9fcda36f141f2a17316736ca78d01e07/values.yaml#L1659
You'll have to create the config file yourself in k8s an then reference it as a volume mount in your workers.:
```
[server]
; configure Google DNS
dns-server=8.8.8.8
dns-server=8.8.4.4
```

Alternatively, using containerd may be simpler for this: https://github.com/concourse/concourse-chart/blob/016fb3ea9fcda36f141f2a17316736ca78d01e07/values.yaml#L1699

If you want to change runtimes to containerd: https://github.com/concourse/concourse-chart/blob/016fb3ea9fcda36f141f2a17316736ca78d01e07/values.yaml#L1618-L1620
and then you can mess around with these settings: https://github.com/concourse/concourse-chart/blob/016fb3ea9fcda36f141f2a17316736ca78d01e07/values.yaml#L1677-L1725

**Reply by gnometsunami** [reply]
Here are a few more errors.

Enabling dns proxy in garden:
`checking origin busybox failed: initialize transport: Get "https://index.docker.io/v2/": read tcp 10.80.0.2:42412->34.193.45.8:443: read: connection reset by peer`

Enabling containerd:
`ERRO[0010] checking origin busybox failed: initialize transport: Get "https://index.docker.io/v2/": net/http: TLS handshake timeout`

I dont think this is a dns issue because the names are being resolved. I think it's a container networking issue. After skimming the values.yaml, ".concourse.worker.http_proxy" is looking suspicious. Do I need to configure a proxy?

**Reply by gnometsunami** [reply]
More debugging seems to suggest that I do not have a dns problem-
```
$ fly -t ci intercept -j hello-world/hello-world-job
bash-5.0$ ping google.com
PING google.com (172.217.14.78): 56 data bytes
64 bytes from 172.217.14.78: seq=0 ttl=116 time=20.466 ms
...
```

**Reply by gnometsunami** [reply]
from fly intercept:
```
bash-5.1# wget https://github.com/concourse/concourse/releases/download/v7.4.0/fly-7.4.0-linux-amd64.tgz
Connecting to github.com (140.82.112.3:443)
ssl_client: github.com: handshake failed: Connection reset by peer
wget: error getting response: Connection reset by peer
```

**Reply by taylorsilva** [reply]
Agreed, definitely not a DNS issue.
Do you get any errors for non-SSL connections?

**Reply by gnometsunami** [reply]
Sure. I couldnt think of any external services that still use http, so I brought up a pod in the concourse namespace. Weird results.
```
kubectl -n concourse run nginx --image=nginx:alpine --port 80
fly -t ci intercept -j hello-world/hello-world-job

bash-5.1# ping 10.233.65.18
PING 10.233.65.18 (10.233.65.18): 56 data bytes
64 bytes from 10.233.65.18: seq=0 ttl=61 time=0.499 ms

bash-5.1# wget http://10.233.65.18/index.html
Connecting to 10.233.65.18 (10.233.65.18:80)
wget: can't connect to remote host (10.233.65.18): Host is unreachable
```

**Reply by taylorsilva** [reply]
That is really strange. Not sure what to make of that :( What k8s engine are you running on? Google's (GKE), AWS's (EKS), Azure's (AKS), etc.
I wonder if the iptables are conflicting with each other somehow, between what k8s and Concourse is configuring?

**Reply by gnometsunami** [reply]
This cluster is vanilla kubeadm, flannel cni.
